### PR TITLE
ITerminalOptions: Make fields required

### LINKED ISSUE #3948: ITerminalOptions: Make fields required
In #3667, all getOptions were replaced by their new equivalent: 
`this.dimensions.scaledCharLeft = Math.floor(this._terminal.getOption('letterSpacing') / 2);`
became
`this.dimensions.scaledCharLeft = Math.floor(this._terminal.options.letterSpacing! / 2);`

Mind the "!" there, which is a shortcut to let typescript know that the variable is for sure defined. In fact, that can be safely assumed since the default parameters are filled in. 
If the class ITerminalOptions is changed to have required fields which are not optional, it would be possible to have 
``this.dimensions.scaledCharLeft = Math.floor(this._terminal.options.letterSpacing / 2);``

It was kind of explored in #3448 that
``options: ITerminalOptions;``
can be changed to 
`get options(): ITerminalOptions;`
`set options(options: Partial<ITerminalOptions>): void;`
which would be an equivalent of the current code. Just ITerminalOptions is changed to have required fields. 

There are some settings though which can be "undefined": 
* fastScrollModifier: Perhaps we can introduce "none" here
* overviewRulerWidth: Use 0 to have this disabled?

This would also make the code more aligned, since the typings have ITerminalOptions with all fields optional and the internal ITerminalOptions have required fields. 

The breaking change part here lies in the fact that addons can use ITerminalOptions as type. If this is passed as a parameter, they would need to change the parameter to Partial\<ITerminalOptions\>.

### PR DESCRIPTION
**Pull request by silamon**
Fixes #3948. Pull request #3982 but the other way around, introducing Required. 

It's still a breaking change though, thanks to the options changes in fastScrollModifier and overviewRulerWidth.

### CODE REVIEW

File: typings/xterm-headless.d.ts [hunk 1]
```diff
@@ -538,11 +538,12 @@ declare module 'xterm-headless' {
...
+     */
+    get options(): Required<ITerminalOptions>;
+
+    /**
+     * Gets or sets the terminal options. This supports setting multiple options.
      *
-     * @example Set a single option
-     * ```typescript
-     * terminal.options.fontSize = 12;
-     * ```
```
Comment (Tyriar) [inline typings/xterm-headless.d.ts:545]: Should we move this example to `get options` now?

**Review by Tyriar — APPROVED** [review]
Looking good 👍

File: typings/xterm-headless.d.ts (hunk 1 above)
Comment (silamon) [reply typings/xterm-headless.d.ts:545]: Yeah, indeed.

**Review by Tyriar — APPROVED** [review]
👍

**Pull request merged**